File: myapp/parent/Step50_LambdaLoopGet.cs
using sfn_ut.sfn.await;

namespace sfn_ut.myapp.parent;

public class Step50_LambdaLoopGet : SfnIsComplete<ParentPayload> {
  public override SfnIsCompleteOutput<ParentPayload> IsComplete(ParentPayload payload) {
    bool isComplete = false;
    
    if (payload.LambdaLoopIteration >= 5)
      isComplete = true;
    else
      payload.LambdaLoopIteration += 1;

    return new SfnIsCompleteOutput<ParentPayload>(payload, isComplete);
  }
}

File: java-sfn-unit-test/src/main/java/com/sfnut/myapp/parent/Step50_LambdaLoopGet.java
package com.sfnut.myapp.parent;

import com.sfnut.sfn.lambdawithloop.SfnIsComplete;
import com.sfnut.sfn.lambdawithloop.SfnIsCompleteOutput;

public class Step50_LambdaLoopGet extends SfnIsComplete<ParentPayload> {
    @Override
    public SfnIsCompleteOutput<ParentPayload> isComplete(ParentPayload payload) {
        boolean isComplete = false;
        
        if (payload.getLambdaLoopIteration() >= 5) {
            isComplete = true;
        } else {
            payload.setLambdaLoopIteration(payload.getLambdaLoopIteration() + 1);
        }

        return new SfnIsCompleteOutput<>(payload, isComplete);
    }
}
